Makefile

```diff
@@ -67,16 +70,28 @@ verify: fmt vet tidy generate ## verification checks against code.
 
 .PHONY: test
 test: manifests generate fmt vet envtest ## Run tests.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./... -coverprofile cover.out
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./util ./controllers -coverprofile cover.out
 
 .PHONY: test-e2e
-test-e2e: ## Run e2e tests.
-	@echo "TODO: PLEASE DEFINE test-e2e TARGET"
+test-e2e: ginkgo ## Run e2e tests.
+	$(GINKGO) -trace -progress test ./test/e2e 
+
+e2e: KIND_CLUSTER_NAME=oria-e2e
+e2e: docker-build kind-cluster kind-load-images deploy test-e2e kind-cluster-cleanup ## Run e2e tests against ephemeral KinD cluster
 
 .PHONY: lint
 lint: golangci-lint ## Run golangci-lint linter
 	$(GOLANGCI_LINT) run
 
+kind-cluster: kind # create a kind cluster
+	$(KIND) create cluster --name ${KIND_CLUSTER_NAME}
+	$(KIND) export kubeconfig --name ${KIND_CLUSTER_NAME}
+
+kind-cluster-cleanup: kind # delete a kind cluster
+	$(KIND) delete cluster --name ${KIND_CLUSTER_NAME}
+
+kind-load-images: kind
+	$(KIND) load docker-image ${IMG} --name ${KIND_CLUSTER_NAME}
```

Comment:
These all need `.PHONY:` lines. Basically Makefile targets are typically "files", since all of these don't represent files, `.PHONY` is required. https://stackoverflow.com/questions/2145590/what-is-the-purpose-of-phony-in-a-makefile

Reply:
Should be fixed with commit https://github.com/operator-framework/oria-operator/commit/9cfafd2158ea14da68bb62915c3defd4878b3ba2